### DISCUSSION TITLE
Voice Prompt with Driving Profile While Navigating: Pedestrian crosswalks

### DISCUSSION THREAD

**Discussion by Zian** [osmandapp/OsmAnd#14068] [Q&A]
When I drove on a road with many controlled intersections with pedestrian crosswalks tagged as crossing=marked & highway=crossing, OsmAnd frequently announced a pedestrian crossing. The announcements were spurious because the crosswalks are at controlled intersections. Pedestrians are expected to push a button and wait for the traffic lights to indicate that it is safe to cross.

Is it normal to receive alerts for pedestrian crossings that are controlled by the intersection's traffic lights?

Example Route:

- From: https://www.openstreetmap.org/way/31817952
- To: https://www.openstreetmap.org/way/31819770

As you drive south on El Camino Real, you will hear an alert shortly before reaching the intersection of El Camino Real and Carmel Valley Road (https://www.openstreetmap.org/node/1774158840).

**Comment by tommycrock** [comment]
The tags should be highway=crossing + crossing=traffic_signals
https://wiki.openstreetmap.org/wiki/Tag:crossing%3Dtraffic_signals
crossing=marked is for ones we'd call a zebra crossing in the UK, where there's clear markings on the road and pedestrians have priority (cars have to stop if they are waiting/crossing).

**Reply by Lee-Carre** [reply]
> `crossing=marked` is for ones we'd call a zebra crossing in the UK

Well, from my reading of the wiki, and judging by the intentions of presets, for a zebra crossing then `crossing=uncontrolled` (`controlled` would be e.g. pelican; traffic signals) is preferred (and more common, I believe).

Few crossings are unmarked (though, some exist), so `crossing=marked` is almost redundant, being the norm.
Though, could always set multiple semicolon-separated values, for exhaustive completeness 🙂.

If I recall, there's `crossing_ref` for specifying the UK-type by name. E.g. `crossing_ref=zebra` in this case.
I believe the idea is that the regular `crossing` tag should be a generic description (not nation-specific or using obscure terminology).

**Reply by Zian** [reply]
I interpreted the wiki the same way as Lee-Carre. 

As a minor matter, I disagree about "few crossing are unmarked". In my mind, each driveway that leads to a street has an unmarked crossing. Imagine a steady stream of people walking along the sidewalk and a parade of cars driving into the driveway. There is a crossing at the point where a car smashes into a pedestrian.

**Comment by Lee-Carre** [comment]
> Is it normal to receive alerts for pedestrian crossings that are controlled by the intersection's traffic lights?

Further to [tommycrock's comment about (correct) tagging of crossings](/osmandapp/OsmAnd/discussions/14068#discussioncomment-2564509) (because map data must be accurate / correct, first, for routing & navigation to function properly); to my understanding, it's a question of how to define the feature when it serves both purposes / functions.

If I were mapping such features, then at an intersection I would tag them as traffic signals, since that's the primary aspect (there just happens to be a crossing there, also).

For non-intersection crossings, it being a crossing is the primary aspect.

Especially since, when both (signals & crossing), one has to decide what to set the `highway` tag (it accepts either value), considering that multiple semicolon-separated values aren't well-supported (yet? 🤞) by much software.

However, with all that said, I can imagine some users waning to be notified about every crossing (since it's a hazard), not only the ones which are between intersections (or junctions, in Brit-speak). So, perhaps the behaviour needs to be user-configurable (another toggle in the profile config). Else, something more elaborate to define exactly which crossing types to notify about.
Note that there's arguably also the difference between on-screen alerts, and voice announcements.

**Reply by thompehw** [reply]
I agree, @Lee-Carre: it's important to make a distinction between (1) a light-controlled junction at which pedestrians can also cross (maybe with their own button to influence the light sequence), and (2) lights that are solely for a pedestrian crossing and *only* stop vehicles when a pedestrian presses the button. 

I would expect routing software to apply a much larger penalty for drivers at (1) than (2).

As I understand it, case (1) needs highway=traffic_signals and case (2) is easily dealt with by highway=crossing + crossing=traffic_signals. Things can be made more sophisticated by mapping more details of the pedestrian way than just the node at which it intersects the road, but I wish there were a simpler way to map cases where pedestrians can use controlled intersections.

At the junction in question it looks from streetview (not to be used for mapping info, of course) that it's case (1). On OSM, footway crossings have already been mapped separately all around this intersection and it seems correct that these trigger OsmAnd's warning.

Navigation_settings>Voice_prompts>Pedestrian_crosswalks can turn these warnings off. In general it's nice to have them, but maybe one could make a duplicate driving profile with them disabled for use in areas where they would be annoying?

**Reply by tommycrock** [reply]
I'm not sure I understand all of your reply @thompehw - when you're saying it's correct OsmAnd triggers the warnings do you mean given how it's tagged or because of the nature of the crossings? I think the nodes where the pedestrians cross the road should be tagged crossing=traffic_signals, assuming there is some interruption in the traffic flow that allows people to cross and tells the people they can cross (I didn't look on streetview). Given the pedestrian ways have been separately mapped I would then also separately map the highway=traffic_signals at the stop lines instead of the junction, but either case should be handled. If there's only an opportunity to cross at a place where there are traffic lights but no signal for pedestrians I would tag those nodes crossing=unmarked, e.g. https://www.openstreetmap.org/node/9006986543

**Reply by Zian** [reply]
Thanks for the tip about the OsmAnd setting, thompehw. I agree with you that there should be a distinction between crossings that are primarily road intersections and crossings that are primarily pedestrian intersections. I've been mapping things the way you described.

However, like tommycrock, I don't think it's correct for OsmAnd to alert about the pedestrian crossings that you saw because they are controlled crossings.
